.travis.yml

```diff
@@ -1,49 +1,92 @@
-  - docker build -f latest/Dockerfile -t nodered/node-red-docker .
-  - docker build -f latest/Dockerfile -t nodered/node-red-docker:v8 --build-arg NODE_VERSION=8 .
-  - docker build -f slim/Dockerfile -t nodered/node-red-docker:slim .
-  - docker build -f slim/Dockerfile -t nodered/node-red-docker:slim-v8 --build-arg NODE_VERSION=8 .
-  - docker build -f rpi/Dockerfile -t nodered/node-red-docker:rpi .
-  - docker build -f rpi/Dockerfile -t nodered/node-red-docker:rpi-v8 --build-arg NODE_VERSION=8 .
-  - docker tag nodered/node-red-docker nodered/node-red-docker:$VERSION
-  - docker tag nodered/node-red-docker nodered/node-red-docker:v6
-  - docker tag nodered/node-red-docker nodered/node-red-docker:$VERSION-v6
-  - docker tag nodered/node-red-docker:v8 nodered/node-red-docker:$VERSION-v8
-  - docker tag nodered/node-red-docker:slim nodered/node-red-docker:$VERSION-slim
-  - docker tag nodered/node-red-docker:slim nodered/node-red-docker:slim-v6
-  - docker tag nodered/node-red-docker:slim nodered/node-red-docker:$VERSION-slim-v6
-  - docker tag nodered/node-red-docker:slim-v8 nodered/node-red-docker:$VERSION-slim-v8
-  - docker tag nodered/node-red-docker:rpi nodered/node-red-docker:$VERSION-rpi
-  - docker tag nodered/node-red-docker:rpi nodered/node-red-docker:rpi-v6
-  - docker tag nodered/node-red-docker:rpi nodered/node-red-docker:$VERSION-rpi-v6
-  - docker tag nodered/node-red-docker:rpi-v8 nodered/node-red-docker:$VERSION-rpi-v8
+  - ./.docker/docker.sh prepare
+
+install: true
+
+before_script:
+    # Set BUILD_VERSION
+    - if [ ! -z "${TRAVIS_TAG}" ]; then export BUILD_VERSION=${TRAVIS_TAG:1}; else export BUILD_VERSION=beta; fi
 
 script:
-  - docker run -d nodered/node-red-docker:latest
-  - docker ps | grep latest
-  - docker run -d nodered/node-red-docker:v8
-  - docker ps | grep v8
-  - docker run -d nodered/node-red-docker:slim
-  - docker ps | grep slim
-  - docker run -d nodered/node-red-docker:slim-v8
-  - docker ps | grep slim-v8
-  - docker run -d nodered/node-red-docker:rpi
-  - docker ps | grep rpi
-  - docker run -d nodered/node-red-docker:rpi-v8
-  - docker ps | grep rpi-v8
-
-after_success:
-  - if [ "$TRAVIS_BRANCH" == "master" -a "$TRAVIS_PULL_REQUEST" == "false" ]; then
-    docker login -u "$DOCKER_USERNAME" -p "$DOCKER_PASSWORD";
-    docker push nodered/node-red-docker;
+  # Update NODE_RED_VERSION from package.json
+  - ./.docker/update_version.sh
+
+  # Build Docker image
+  - ./.docker/docker.sh build
+
+  # Test Docker image
+  - ./.docker/docker.sh test
+
+  # Push Docker image
+  - >
+    if [ ! -z "${TRAVIS_TAG}" ]; then
+      # Tag Docker image
+      ./.docker/docker.sh tag
+
+      # Docker Login
+      echo "${DOCKER_PASSWORD}" | docker login -u "${DOCKER_USERNAME}" --password-stdin
+
+      # Push Docker image
+      ./.docker/docker.sh push
+
+      # Docker Logout
+      docker logout
     fi
+
+jobs:
+    include:
+        - stage: manifest
+          if: tag =~ ^v
```

Comment:
Am I right to read this as meaning it will only push to dockerhub when we push a tag?

Reply:
I'll explain later today.

Reply:
@knolleary, i've added some comments to explain, your question.